### PR TITLE
Dhcp statistics metrics

### LINKED ISSUE #47: Collect dhcp service statistics metrics
Collect DHCP service and DHCP IP pool statistic metrics including:
 - Number of ack packets
 - Number of decline packets
 - Number of discover packets
 - Number of errors packets
 - Number of informs packets
 - Number of nack packets
 - Number of offer packets
 - Number of release packets
 - Number of request packets
 - IP Pool number of allocated IP
 - IP Pool number of total IP

### PR DESCRIPTION
**Pull request by walbertus**
Collect DHCP service and DHCP IP pool statistic metrics including:
 - Number of ack packets
 - Number of decline packets
 - Number of discover packets
 - Number of errors packets
 - Number of informs packets
 - Number of nack packets
 - Number of offer packets
 - Number of release packets
 - Number of request packets
 - IP Pool number of allocated IP
 - IP Pool number of total IP

In the process also refactor:
 - Extract method to list all dhcp server from dhcp status metrics
 - Abstract out the method to list all dhcp server in nsxt client
 - Update dhcp collector test to fit new structure

Signed-off-by: William Albertus Dembo <w.albertusd@gmail.com>

### CODE REVIEW

**Comment by walbertus** [comment]
Closes: #49

File: collector/dhcp_collector.go [hunk 1]
```diff
@@ -40,44 +57,139 @@ func newDHCPCollector(dhcpClient client.DHCPClient, logger log.Logger) *dhcpColl
...
+	dhcpIPPoolSize := prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, "dhcp", "ip_pool_size_total"),
+		"Total size of dhcp ip pool",
+		[]string{"id", "dhcp_id"},
+		nil,
+	)
+	dhcpIPPoolAllocated := prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, "dhcp", "ip_pool_allocated_total"),
+		"Total number of allocated ip of dhcp ip pool",
+		[]string{"id", "dhcp_id"},
```
Comment (girikuncoro) [inline collector/dhcp_collector.go:123]: should we also include pool id and ip pool range (start, end) if provided?

Comment (girikuncoro) [reply collector/dhcp_collector.go:123]: oh nvm just saw this id is actually ip_pool_id

File: collector/dhcp_collector_test.go [hunk 2]
```diff
@@ -12,35 +12,24 @@ import (
...
-	dhcpListError error
+	responses []mockDHCPResponse
 }
 
 type mockDHCPResponse struct {
 	ID          string
 	DisplayName string
 	Status      string
 	Error       error
+	Statistic   manager.DhcpStatistics
```
Comment (girikuncoro) [inline collector/dhcp_collector_test.go]: nit: Statistics
```suggestion
	Statistics   manager.DhcpStatistics
```

**Review by girikuncoro — APPROVED** [review]
looks good!
/lgtm

File: collector/dhcp_collector.go (hunk 1 above)
Comment (walbertus) [reply collector/dhcp_collector.go:123]: It's not provided in statistics

**Pull request merged**